ibis/common/grounds.py

```diff
@@ -120,19 +125,20 @@ def __recreate__(cls, kwargs: Any) -> Self:
         kwargs = cls.__signature__.validate_nobind(**kwargs)
         return super().__create__(**kwargs)
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
+    def __init__(self, **kwargs: Any) -> None:
         # set the already validated arguments
         for name, value in kwargs.items():
             object.__setattr__(self, name, value)
-
-        # post-initialize the remaining attributes
+        # initialize the remaining attributes
         for name, field in self.__attributes__.items():
-            if isinstance(field, Attribute):
-                if (value := field.initialize(self)) is not EMPTY:
-                    object.__setattr__(self, name, value)
+            if (default := field.initialize(self)) is not EMPTY:
+                object.__setattr__(self, name, default)
 
     def __setattr__(self, name, value) -> None:
-        if field := self.__attributes__.get(name):
+        # first try to look up the argument then the attribute
```

Comment:
This is mostly used by the `Config` global object since that is the only mutable `Annotable` subclass I'm aware of.